### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by timiimit**
- Clean up code
- check some TODO's off the list
- Add newly discovered endpoints
- Split `UnrealTournamentGameController` into 3 controllers

### CODE REVIEW

File: UT4MasterServer/Controllers/SessionController.cs [hunk 1]
```diff
@@ -49,121 +48,123 @@ public class SessionController : JsonAPIController
...
-		if (session == null) // only here to prevent null warnings, should never happen
+		if (session == null)
 		{
-			var message = $"Empty session. grant_type: {request.GrantType}";
+			var message = $"Invalid credentials";
 			logger.LogError(message);
-			return BadRequest(message);
+
+			// TODO: Find proper response
+			return BadRequest(new ErrorResponse
```
Comment (Saibamen) [inline UT4MasterServer/Controllers/SessionController.cs:158]: `return Unauthorized(new ErrorResponse` (see https://github.com/timiimit/UT4MasterServer/pull/44#discussion_r1061518990)

File: UT4MasterServer/Controllers/SessionController.cs [hunk 2]
```diff
@@ -49,121 +48,123 @@ public class SessionController : JsonAPIController
...
+					Error = "unsupported_grant_type",
+				});
+			}
 		}
 
-		if (session == null) // only here to prevent null warnings, should never happen
+		if (session == null)
 		{
-			var message = $"Empty session. grant_type: {request.GrantType}";
+			var message = $"Invalid credentials";
```
Comment (Saibamen) [inline UT4MasterServer/Controllers/SessionController.cs:154]: Are you sure this is only for `password` grant_type?

File: .Tests/XUnit.Tests/GameServerAttributeTest.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 namespace XUnit.Tests;
 
+#if false
 public class GameServerAttributeTest
```
Comment (Saibamen) [inline .Tests/XUnit.Tests/GameServerAttributeTest.cs:8]: Use Skip for xUnit tests - we will see which tests was skipped + you can also add message

File: UT4MasterServer/Controllers/SessionController.cs (hunk 2 above)
Comment (timiimit) [reply UT4MasterServer/Controllers/SessionController.cs:154]: credentials is any info used to authenticate a user. this includes `exchange_code` for `grant_type=exchange_code`, `username` and `password` for `grant_type=password`, etc.

File: .Tests/XUnit.Tests/GameServerAttributeTest.cs (hunk 3 above)
Comment (timiimit) [reply .Tests/XUnit.Tests/GameServerAttributeTest.cs:8]: I remove all code which this class was testing. It is just compiler errors now. i should just delete the whole file tbh.

File: UT4MasterServer/Controllers/SessionController.cs (hunk 1 above)
Comment (timiimit) [reply UT4MasterServer/Controllers/SessionController.cs:158]: oops, I'll put it back to `Unauthorized` in another PR.

**Pull request merged**